app/src/main/java/de/westnordost/streetcomplete/quests/address/AddAddressStreet.kt

```diff
@@ -17,8 +17,8 @@ class AddAddressStreet : OsmElementQuestType<StreetOrPlaceName> {
 
     private val filter by lazy { """
         nodes, ways, relations with
-          (addr:housenumber or addr:housename) and !addr:street and !addr:place and !addr:block_number
-          or addr:streetnumber and !addr:street
+          (addr:housenumber or addr:housename) and !addr:street and !addr:place and !addr:block_number and !addr:substreet and !addr:parentstreet
+          or addr:streetnumber and !addr:street and !addr:substreet and !addr:parentstreet
```

Comment:
Could be refactored to

```suggestion
          !addr:street and (
            (addr:housenumber or addr:housename) and !addr:place and !addr:block_number
            or addr:streetnumber
          ) and !addr:substreet and !addr:parentstreet
```

But I don't know if that's easier to understand.

Reply:
I think it is fine this way